esp/esp/program/models/__init__.py

```diff
@@ -1315,10 +1323,17 @@ def save(self, *args, **kwargs):
     @cache_function
     def getLastForProgram(user, program):
         """ Returns the newest RegistrationProfile attached to this user and this program (or any ancestor of this program). """
-        if user.is_anonymous():
-            regProfList = RegistrationProfile.objects.none()
-        else:
-            regProfList = (RegistrationProfile.objects
+        has_access = Permission.user_has_perm(user, '/Profile', program)
+        registration_profile = None
```

Comment:
This line probable isn't necessary since it's set in both branches of the if statement below.